ang/civicase-base/providers/case-type-category.provider.js

```diff
@@ -8,6 +8,8 @@
    */
   function CaseTypeCategoryProvider () {
     var allCaseTypeCategories = CRM['civicase-base'].caseTypeCategories;
+    var caseTypeCategoriesWithAccessToActivities =
+      CRM['civicase-base'].caseTypeCategoriesWithAccessToActivities;
```

Comment:
This is not possible, because inside a provider you cant access any other angular component like constants, specially in the Constructor function.

Reply:
Ok, then can we at least pass `CRM['civicase-base']` to the wrapper function?

```js
(function (..., civicaseSettings) {
  // ...
  var caseTypeCategoriesWithAccessToActivities = civicaseSettings.caseTypeCategoriesWithAccessToActivities;
  // ...
})(..., CRM['civicase-base']);
```

Reply:
We should try to pass these type of dependencies through angular services/constant/providers as much as possible for several reasons:

1 - We don't rely on a global value directly. If we need to reference this value on different parts of our app, they will reference an angular provider, and only the provider will reference the global value.
2 - We can override angular providers. This is good for testing, and also for other extensions. For example, the `currentCaseCategory` value could be overridden. 
3 - If we need to define defaults for this value (in case it's missing from the global object) we define this default in the provider and don't have to define it each time we reference this value.

Providers can access other providers/services/etc by doing something similar to this:

```js
module.provider('myProvider', function (myOtherValueProvider) {
  // either this:
 var myOtherValue = myOtherValueProvider.getThisMagicalValue();

  // or this:
  var myOtherValue = myotherValueProvider.$get(listOfDependencies);
});
```

Key points:

* A provider can include the provider for another value if they add the `Provider` subfix to it.
* The provider can define a way to pass the value through the provider, like with the `getThisMagicalValue` example method.
* The provider can also return the value through the `$get` function that is mandatory for providers, but we need to pass any dependencies that it requires.

There's more boilerplate, but this could be worth it in a lot of scenarios. We could discuss this point as well.

Reply:
Right, but not sure if this will be any better. Because we are only using global values inside providers, because its not possible use other angular components.
Now the only option is to use another angular provider, but then it would be the same thing right? because the other provider will be using global values.
As in the end we would need to use the global values, and we cant use constants, it does not make sense to create additional providers. Is not that the case?

Reply:
You are absolutely right @deb1990 . It is too much to create a provider just for that. But I noticed we are still using the old name for it. Can't we name it `caseTypeCategoriesWhereUserCanAccessActivities`?